### PR TITLE
Update arcade (2.9.x branch)

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Dependencies>
+  <ProductDependencies>
+    <Dependency Name="Microsoft.NETCore.Compilers" Version="3.7.0-3.20269.11">
```
Comment (sharwell) [inline eng/Version.Details.xml:4]: ❔ Should this also be set back to 3.3.1-beta3-final?

Comment (mavasani) [reply eng/Version.Details.xml:4]: @dotpaul to decide. I think it seems reasonable to move to latest compiler if nullable warnings are easy to fix.

Comment (dotpaul) [reply eng/Version.Details.xml:4]: I'd say do whatever is easiest for this PR to upgrade arcade, assuming it's not hard to move to the latest compilers in a separate PR.

**Comment by mavasani** [comment]
Thanks a lot @sharwell!

**Pull request merged**